TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ultimategoal/util/ModuleExecutor.java

```diff
@@ -37,8 +37,11 @@ public void run() {
                 lastUpdateTime = SystemClock.elapsedRealtime() - startTime;
             }
 
-            if (robot.isStopRequested() && robot.WILL_FILE_DUMP) {
-                robot.fileDump.writeFilesToDevice();
+            if (robot.WILL_FILE_DUMP) {
+                robot.fileDump.doTick();
+
+                if (robot.isStopRequested())
```

Comment:
Maybe add a stopping case to Robot that is called at the end of every op mode, to ensure we don't miss any closing stuff. I'm surprised we never got a problem even though we never .join() our ModuleExecutor thread.